packs/st2cd/actions/tag_branch.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+GIT=`which git`
+REPO=$1
+DATE=`date +%s`
+BRANCH=$2
+OUTPUT=/tmp/git-output-$DATE
+
+cd ${REPO}
+OUT=`$GIT checkout ${BRANCH} -q && $GIT pull origin ${BRANCH} -q > ${OUTPUT}`
+if [[ $? == 0 ]]
+then
+  NEWTAG=`git describe --tags --abbrev=0 | awk -F '[ .]' '{print $1"."$2"."$3+1}'`
+  if [[ $? == 0 ]]
+  then
+    OUT=`$GIT tag -a ${NEWTAG} -m "Creating tag ${NEWTAG} for branch ${BRANCH}" && $GIT push origin ${NEWTAG} -q`
```

Comment:
> We can add it as a parameter to the all previous workflows and pass it through. Or write it to the kvstore...maybe execution_id:SHA or something.

I thought about the kvstore and I don't think it will work because we don't have a context identifier across workflows. Passing the SHA as a parameter seems to be the only option. We can do that.